spoc/io.py

```diff
@@ -73,12 +104,149 @@ def load_chromosome_sizes(path: str):
+
+        """
+        metadata = self._load_metadata(path)
+        # find matching pixels
+        for pixel_path, value in metadata.items():
+            param_value = PixelParameters(**value)
+            if param_value == global_parameters:
+                selected_value = param_value
+                break
+        else:
+            raise ValueError(f"No matching pixels found for {global_parameters}")
+        # rewrite path to contain parent folder
+        pixel_path = Path(path) / pixel_path
+        if load_dataframe:
+            df = self._parquet_reader_func(pixel_path)
+        else:
+            df = pixel_path
+        return Pixels(df, **selected_value.dict())
+
+    def load_contacts(
+        self, path: str, global_parameters: ContactsParameters
+    ) -> Contacts:
+        """Loads specific contacts instance based on global parameters.
+        load_dataframe specifies whether the dataframe should be loaded"""
+        metadata = self._load_metadata(path)
+        # find matching contacts
+        for contacts_path, value in metadata.items():
+            param_value = ContactsParameters(**value)
+            if param_value == global_parameters:
+                selected_value = param_value
+                break
+        else:
+            raise ValueError(f"No matching contacts found for {global_parameters}")
+        # rewrite path to contain parent folder
+        contacts_path = Path(path) / contacts_path
+        df = self._parquet_reader_func(contacts_path)
+        return Contacts(df, **selected_value.dict())
 
     @staticmethod
-    def write_pixels(path: str, pixels: Pixels) -> None:
-        """Write pixels"""
-        pixels.data.to_parquet(path, row_group_size=1024*1024)
+    def _get_object_hash_path(path: str, data_object: Union[Pixels, Contacts]) -> str:
+        hash_string = path + json.dumps(data_object.get_global_parameters().dict())
+        return md5(hash_string.encode()).hexdigest() + ".parquet"
+
+    def write_pixels(self, path: str, pixels: Pixels) -> None:
+        """Write pixels
+
+        Args:
+            path (str): Path to write the pixel data to.
+            pixels (Pixels): Pixels object containing the pixel data.
+
+        Returns:
+            None
+
+        """
+        # check whether path exists
+        metadata_path = Path(path) / "metadata.json"
+        if not Path(path).exists():
+            # we need to create the directory first
+            os.mkdir(path)
+            current_metadata = {}
+        else:
+            current_metadata = FileManager._load_metadata(path)
+        # create new file path -> hash of directory path and parameters
+        write_path = Path(path) / self._get_object_hash_path(path, pixels)
+        # write pixels
+        if pixels.data is None:
+            raise ValueError(
+                "Writing pixels only suppported for pixels hodling dataframes!"
+            )
+        pixels.data.to_parquet(write_path, row_group_size=1024 * 1024)
```

Comment:
in the latest pyarrow version the actually enforced this default `row_group_size`, but I suggest we keep specifying it in our code anyway.